scipy/signal/tests/test_splines.py

```diff
@@ -367,13 +366,15 @@ def test_symiir2(self, dtype, precision, xp):
         out = symiirorder2(signal, r, omega, precision)
         xp_assert_close(out, exp, atol=4e-6, rtol=6e-7)
 
-    @skip_xp_backends(cpu_only=True)
+    @skip_xp_backends(cpu_only=True, exceptions=["cupy"], reason="C internals")
     @pytest.mark.parametrize('dtyp', ['float32', 'float64'])
     def test_symiir2_values(self, dtyp, xp):
         rng = np.random.RandomState(1234)
         s = rng.uniform(size=16).astype(dtyp)
         s = xp.asarray(s)
-        dtyp = getattr(xp, dtyp)
+
+        # cupy returns f64 for f32 inputs
```

Comment:
This one is a bit funny: I'm pretty sure CuPy's version was made to follow SciPy's, and now we're adjusting SciPy's to match CuPy's :-).